Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -164,14 +167,31 @@ extension Channel {
                                        initialLocalSettings: [HTTP2Setting] = nioDefaultSettings,
                                        position: ChannelPipeline.Position = .last,
                                        targetWindowSize: Int = 65535,
-                                       inboundStreamInitializer: ((Channel) -> EventLoopFuture<Void>)?) -> EventLoopFuture<HTTP2StreamMultiplexer> {
+                                       inboundStreamInitializer: NIOChannelInitializer?) -> EventLoopFuture<HTTP2StreamMultiplexer> {
         var handlers = [ChannelHandler]()
         handlers.reserveCapacity(2)  // Update this if we need to add more handlers, to avoid unnecessary reallocation.
         handlers.append(NIOHTTP2Handler(mode: mode, initialSettings: initialLocalSettings))
         let multiplexer = HTTP2StreamMultiplexer(mode: mode, channel: self, targetWindowSize: targetWindowSize, inboundStreamInitializer: inboundStreamInitializer)
         handlers.append(multiplexer)
 
-        return self.pipeline.addHandlers(handlers, position: position).map { multiplexer }
+        do {
+            if self.eventLoop.inEventLoop {
+                try self.pipeline.syncOperations.addHandlers(handlers, position: position)
+                return self.eventLoop.makeSucceededFuture(multiplexer)
+            } else {
+                let loopBoundMultiplexer = NIOLoopBound(multiplexer, eventLoop: self.eventLoop)
+                let loopBoundHandlers = NIOLoopBound(handlers, eventLoop: self.eventLoop)
+                let loopBoundPosition = NIOLoopBound(position, eventLoop: self.eventLoop)
```

Comment:
We know we're not on the EL here so these will fail preconditions on init.

Reply:
Rather than moving the handlers onto the EL we can just create them on the EL. The pipeline position is super awkward though because it isn't `Sendable`. I'm not sure what the solution there is (this is a problem NIO needs to solve).

Reply:
I've moved the handler creation but left the warning for this and other use of `position` rather than trying to hide it because it truly is not sendable.